app/components/PlacePopup.tsx

```diff
@@ -44,20 +41,18 @@ const PlacePopup = ({ map, place, show, onClose }: PopupProps) => {
       .setLngLat(coordinates.current)
       .setDOMContent(popupContentRef.current as Node);
 
+    if (show) {
+      popupRef.current.addTo(map);
+      if (zoomToFeature) {
+        map.flyTo({ center: coordinates.current, zoom: 15 });
+      }
+    }
+
     return () => {
       popupRef.current?.remove();
-      popupRef.current = undefined;
+      popupRef.current = null;
     };
-  }, [map, place]);
-
-  useEffect(() => {
-    if (show && map) {
-      popupRef.current?.addTo(map);
-      map.flyTo({ center: coordinates.current, zoom: 15 });
-    } else {
-      popupRef.current?.remove();
-    }
-  }, [show, map]);
+  }, [map, place, show, zoomToFeature]);
 
   return (
     <div ref={popupContentRef}>
```

Comment:
This works, but it causes an error - something about not being able to remove child from node. I think it has something to do with the popup component trying to remove itself from the map, but the GeoSearch removes the component from the DOM when `activePlace` becomes `undefined`. Take a look the [RelatedPlaces](https://github.com/ecds/georgia-coast-atlas-frontend/blob/dev/app/components/RelatedPlaces.tsx#L171-L191) component. It makes a separate instance of the PlacePopup component for each place ("hit" in this context). The `show` property is set by checking if `activePlace` matches the `place` for the PlacePopup. You'll probably have to figure out a different way to match what is clicked on the map with the hit.